src/mpl/include/mpl_thread.h

```diff
@@ -21,15 +21,24 @@
 #define MPL_THREAD_PRIO_HIGH 0
 #define MPL_THREAD_PRIO_LOW 1
 
+/* Use direct compiler keyword if MPL_TLS is defined;
+ * Otherwise, use thread library functions.
+ */
+#if defined MPL_COMPILER_TLS    /* compiler tls support */
+#define MPL_TLS MPL_COMPILER_TLS
+#endif
+
 #if defined(MPL_THREAD_PACKAGE_NAME) && (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_POSIX || MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_UTI)
 #include "mpl_thread_posix.h"
 #elif defined(MPL_THREAD_PACKAGE_NAME) && (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_SOLARIS)
 #include "mpl_thread_solaris.h"
 #elif defined(MPL_THREAD_PACKAGE_NAME) && (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_WIN)
 #include "mpl_thread_win.h"
 #elif defined(MPL_THREAD_PACKAGE_NAME) && (MPL_THREAD_PACKAGE_NAME == MPL_THREAD_PACKAGE_ARGOBOTS)
+#undef MPL_TLS  /* compiler tls cannot be used */
```

Comment:
Let's define it in `mpl_thread_argobots.h`.

Reply:
Don't think it is better not to spread the logic component across multiple files?

Reply:
Updated.

Reply:
Looks good to me!